File: VitalSphere/VitalSphere.Services/Database/Product.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace VitalSphere.Services.Database
{
    public class Product
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(200)]
        public string Name { get; set; } = string.Empty;
        
        [Required]
        public decimal Price { get; set; }
        
        public byte[]? Picture { get; set; }
        public bool IsActive { get; set; } = true;

        [Required]
        public int ProductSubcategoryId { get; set; }
        public ProductSubcategory ProductSubcategory { get; set; } = null!;

        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
}

File: VitalSphere/UI/vital_sphere_mobile/lib/model/product.dart
import 'package:json_annotation/json_annotation.dart';

part 'product.g.dart';

@JsonSerializable()
class Product {
  final int id;
  final String name;
  final double price;
  final String? picture;
  final bool isActive;
  final DateTime createdAt;
  final int productCategoryId;
  final String productCategoryName;
  final int brandId;
  final String brandName;

  Product({
    this.id = 0,
    this.name = '',
    this.price = 0.0,
    this.picture,
    this.isActive = true,
    required this.createdAt,
    this.productCategoryId = 0,
    this.productCategoryName = '',
    this.brandId = 0,
    this.brandName = '',
  });

  factory Product.fromJson(Map<String, dynamic> json) =>
      _$ProductFromJson(json);
  Map<String, dynamic> toJson() => _$ProductToJson(this);
}

